lib/progress.ts

```diff
@@ -84,3 +84,51 @@ export interface IPublishProgress {
    */
   abort(): void;
 }
+
+class GlobalOutputHandler {
+  private progressListener: IPublishProgressListener | undefined;
+  private completionProgress: number;
+
+  constructor(completionProgress: number = 0, progressListener?: IPublishProgressListener) {
+    this.progressListener = progressListener;
+    this.completionProgress = completionProgress;
+  }
+
+  public setListener(listener: IPublishProgressListener) {
+    this.progressListener = listener;
+  }
+
+  public setCompletionProgress(progress: number) {
+    this.completionProgress = progress;
+  }
+
+  public publishEvent(eventType: EventType = EventType.DEBUG, text: string) {
+    const progressEvent: IPublishProgress = {
+      message: text,
+      abort: () => {},
+      percentComplete: this.completionProgress,
+    };
+    // if no listener is passed we just swallow everything.
+    if (this.progressListener) {
+      this.progressListener.onPublishEvent(eventType, progressEvent);
+    }
+  }
+
+  public verbose(text: string) {
+    this.publishEvent(EventType.DEBUG, text);
+  }
+
+  public error(text: string) {
+    this.publishEvent(EventType.FAIL, text);
+  }
+
+  public info(text: string) {
+    this.publishEvent(EventType.SUCCESS, text);
+  }
+
+  public hasListener() {
+    return this.progressListener !== undefined;
+  }
+}
+
+export let globalOutputHandler = new GlobalOutputHandler();
```

Comment:
Executing code on module load is not a great idea. Sure, right now all this code does is instantiate an empty object, but this can easily turn more heavy over time and either slow down or completely crash during import. 

Instead, you can use the singleton pattern so that this code is only executed (once) whenever the first consumer needs it. That access pattern would be something like:

```ts

// for bits of the code that only read it
GlobalOutputHandler.get().verbose(...)

// for bits of the code that create and read it
GlobalOutputHandler.getOrCreate(listener)
```